### ISSUE TITLE
Adjust type definition for createReducer

### ISSUE THREAD

**Issue by olee** [mhoyer/redux-typed-ducks#7]
Just started using this library today and really like how simple it is.

Currently createReducer returns a reducer where the state cannot be undefined.
However latest redux with better typings requires reducers to accept undefined state (to return initialState).

So [this](https://github.com/mhoyer/redux-typed-ducks/blob/master/lib/createReducer.ts#L4) should be changed from
```typescript
(state: TState, action: Action<any>): TState;
```
to
```typescript
(state: TState | undefined, action: Action<any>): TState;
```

Right now I need to do this:
```typescript
import { Reducer } from 'redux';
export const reducer = createReducer<ProductStore>(ducks, initialState) as Reducer<ProductStore>;
```

**Comment by mhoyer** [comment]
@olee, thank you for your help. I am currently on vacation and will check it in two weeks.

**Comment by mhoyer** [comment]
Fixed with #8.
